Documentation/install/upgrade.rst

```diff
@@ -13,30 +13,96 @@ Upgrade Guide
+
+* A ``Secret`` resource: describes the credentials use access the etcd kvstore,
+  if required.
+
+If you have followed the installation guide from :ref:`ds_deploy`, all of the
+above resources were installed via a single ``cilium.yaml`` file.
+
+All upgrades require at least updating the ``DaemonSet`` to point to the newer
+Cilium image tag. However, *safely* upgrading Cilium may also required changes
+additional changes to the ``DaemonSet``, ``ConfigMap`` or ``RBAC`` related
+resources. This depends on your current and target Cilium versions, so it is
+critical to read the :ref:`specific_upgrade` below referring to your target
+Cilium version.
+
+In general, the easiest way to ensure you are making all required updates to
+Cilium related resources is to download new versions of those resources, and
+apply them to your Kubernetes environment.  The recommended high-level workflow
+is:
+
+#. Download a new ``cilium-cm.yaml`` file associated with your target version
+   of Cilium, manually edit the file with any configuration options specific to
+   your environment, and then apply the file to your cluster using kubectl.
+
+#. Update the ``ServiceAccount``, ``ClusterRole``, and ``ClusterRoleBindings``
+   resources by using kubectl to apply a new ``cilium-rbac.yaml`` associated
+   with your target version of Cilium.
+
+#. Update the DaemonSet resource by applying the ``cilium-ds.yaml`` associated
+   with your target version of Cilium.
+
+If there are no changes required to the resources between two Cilium versions
+(e.g., between two patch releases in the same minor version), then it is
+possible to upgrade Cilium simply by editing the cilium image tag in the
+DaemonSet. However, this short-cut should only be done if the
+:ref:`specific_upgrade` instructions below confirm that it is safe.
+
+When upgrading from one minor release to another minor release, for example
+1.x to 1.y, it is generally safer to upgrade first to the latest release in the
+1.x.z series, then subsequently upgrade to 1.y. This way, if there is any
+unexpected issue during the upgrade, it can be safely rolled back to the latest
+good version.
+
+Below we will show examples of how Cilium should be upgraded using Kubernetes
+rolling upgrade functionality in order to preserve any existing Cilium
+configuration changes (e.g., etc configuration) and minimize network
+disruptions for running workloads. These instructions upgrade Cilium to version
+"\ |SCM_BRANCH|" by updating the ``ConfigMap``, ``RBAC`` rules and
+``DaemonSet`` files separately. Rather than installing all configuration in one
+``cilium.yaml`` file, which could override any custom ``ConfigMap``
+configuration, installing these files separately allows upgrade to be staged
+and for user configuration to not be affected by the upgrade.
+
+.. _upgrade_cm:
+
+Upgrade ConfigMap (Recommended)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Upgrading the `ConfigMap` should be done manually before upgrading the ``RBAC``
```

Comment:
Is use one backtick vs. two intentionally done for wrapping ConfigMap vs. other Kubernetes constructs in this section?